File: rb/features.rb
# CaliforniaStatePortal SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CaliforniaStatePortalFeatures
  def self.make_feature(name)
    case name
    when "base"
      CaliforniaStatePortalBaseFeature.new
    when "test"
      CaliforniaStatePortalTestFeature.new
    else
      CaliforniaStatePortalBaseFeature.new
    end
  end
end

File: py/features.py
# CaliforniaStatePortal SDK feature factory

from feature.base_feature import CaliforniaStatePortalBaseFeature
from feature.test_feature import CaliforniaStatePortalTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CaliforniaStatePortalBaseFeature(),
        "test": lambda: CaliforniaStatePortalTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
